packages/core/src/shared/clients/chatDb/chatDb.ts

```diff
@@ -0,0 +1,241 @@
+/*!
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+import Loki from 'lokijs'
+import * as vscode from 'vscode'
+import { TabType } from '../../../amazonq/webview/ui/storages/tabsStorage'
+import { ChatItem, ChatItemType, DetailedListItemGroup } from '@aws/mynah-ui'
+import {
+    ClientType,
+    Conversation,
+    FileSystemAdapter,
+    groupTabsByDate,
+    Tab,
+    TabCollection,
+    updateOrCreateConversation,
+} from './helpers'
+import crypto from 'crypto'
+import path from 'path'
+import { fs } from '../../fs/fs'
+
+export class Database {
+    private static instance: Database | undefined = undefined
+    private db: Loki
+    /**
+     * Keep track of which open tabs have a corresponding history entry. Maps tabIds to historyIds
+     */
+    private historyIdMapping: Map<string, string> = new Map()
+    private dbDirectory: string
+    initialized: boolean = false
+
+    constructor() {
+        this.dbDirectory = path.join(fs.getUserHomeDir(), '.aws', 'amazonq', 'history')
```

Comment:
It is never required to join() string literals. Not on Windows, nor anywhere else. "/" slashes work fine.

```suggestion
        this.dbDirectory = path.join(fs.getUserHomeDir(), '.aws/amazonq/history')
```